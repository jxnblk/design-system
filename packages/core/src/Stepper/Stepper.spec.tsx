import React from 'react'

import { Stepper } from '..'

const children = (
  <React.Fragment>
    <Stepper.Step completed>First Step</Stepper.Step>
    <Stepper.Step active>Second Step</Stepper.Step>
    <Stepper.Step>Third Step</Stepper.Step>
  </React.Fragment>
)

describe('Stepper', () => {
  test('renders without crashing', () => {
    const json = rendererCreateWithTheme(<Stepper />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with children', () => {
    const json = rendererCreateWithTheme(<Stepper>{children}</Stepper>).toJSON()
    expect(json).toMatchSnapshot()
  })
})
