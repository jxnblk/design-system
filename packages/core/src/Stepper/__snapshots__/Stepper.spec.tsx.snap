// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper renders with children 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  color: #0068ef;
  font-size: 14px;
}

.c9 {
  color: #0068ef;
  font-weight: 700;
  font-size: 14px;
}

.c10 {
  color: #4f6f8f;
  font-size: 14px;
}

.c6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #0068ef;
  margin-right: 4px;
  width: 16px;
}

.c7 {
  outline: none;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 9999px;
  border-width: 0;
  border-style: solid;
  border-radius: 9999px;
  font-size: 14px;
  padding: 9.5px 18px;
  background-color: #0068ef;
  color: #fff;
  margin-right: 32px;
}

.c3:hover {
  background-color: #049;
  color: #fff;
}

.c3:focus {
  outline: 0px solid #049;
  box-shadow: 0 0 0 2px #049;
}

.c3:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c4:hover {
  background-color: #fff;
}

.c2 > :not(:last-child) {
  margin-right: 32pxpx;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <button
      className="c3 c4"
      color="primary"
      size="medium"
    >
      <div
        className="c5"
      >
        <svg
          aria-hidden={false}
          aria-label="Step Completed"
          className="c6 c7"
          color="primary"
          fill="currentcolor"
          focusable={false}
          height={16}
          role="img"
          tabIndex={-1}
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.6 15.6l-4.2-4.2L3 12.8l5.6 5.6 12-12L19.2 5 8.6 15.6z"
          />
        </svg>
        <div
          aria-selected={false}
          className="c8"
          color="primary"
          fontSize={1}
        >
          First Step
        </div>
      </div>
    </button>
    <button
      className="c3 c4"
      color="primary"
      size="medium"
    >
      <div
        className="c5"
      >
        <div
          aria-selected={true}
          className="c9"
          color="primary"
          fontSize={1}
        >
          Second Step
        </div>
      </div>
    </button>
    <button
      className="c3 c4"
      color="primary"
      size="medium"
    >
      <div
        className="c5"
      >
        <div
          aria-selected={false}
          className="c10"
          color="text.light"
          fontSize={1}
        >
          Third Step
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`Stepper renders without crashing 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 > :not(:last-child) {
  margin-right: 32pxpx;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
</div>
`;
